base/abstractarray.jl

```diff
@@ -488,7 +488,7 @@ first(itr, n::Integer) = collect(Iterators.take(itr, n))
 # Faster method for vectors
 function first(v::AbstractVector, n::Integer)
     n < 0 && throw(ArgumentError("Number of elements must be nonnegative"))
-    @inbounds v[begin:min(begin + n - 1, end)]
+    v[range(begin, length=min(n, length(v)))]
```

Comment:
If we just want to avoid all possible overflow, `checked_length`/`checked_add`/`checked_sub` seems more intuitive? As `length` itself might have overflowed
```julia
julia> typemin(Int):typemax(Int) |> Base.checked_length
ERROR: OverflowError: 9223372036854775807 - -9223372036854775808 overflowed for type Int64
Stacktrace:
 [1] throw_overflowerr_binaryop(op::Symbol, x::Int64, y::Int64)
   @ Base.Checked .\checked.jl:163
 [2] checked_sub
   @ .\checked.jl:232 [inlined]
 [3] checked_length
   @ .\range.jl:719 [inlined]
 [4] |>(x::UnitRange{Int64}, f::typeof(Base.checked_length))
   @ Base .\operators.jl:907
 [5] top-level scope
   @ REPL[6]:1

julia> typemin(Int):typemax(Int) |> length
0
```

Reply:
Updated now to use `checked_length`